support/BRSyncMode.h

```diff
@@ -33,6 +33,62 @@
 extern "C" {
 #endif
 
+typedef enum {
+    SYNC_STOPPED_REASON_COMPLETE,
+    SYNC_STOPPED_REASON_REQUESTED,
+    SYNC_STOPPED_REASON_UNKNOWN,
+    SYNC_STOPPED_REASON_POSIX
+} BRSyncStoppedReasonType;
+
+typedef struct {
+    BRSyncStoppedReasonType type;
+    union {
+        struct {
+            int errnum;
+            char message[128 + 1];
+        } posix;
+    } u;
+} BRSyncStoppedReason;
+
+extern BRSyncStoppedReason
+BRSyncStoppedReasonComplete(void);
+
+extern BRSyncStoppedReason
+BRSyncStoppedReasonRequested(void);
+
+extern BRSyncStoppedReason
+BRSyncStoppedReasonUnknown(void);
+
+extern BRSyncStoppedReason
+BRSyncStoppedReasonPosix(int errnum,
+                         const char *message);
+
+typedef enum {
+    DISCONNECT_REASON_REQUESTED,
+    DISCONNECT_REASON_UNKNOWN,
+    DISCONNECT_REASON_POSIX
+} BRDisconnectReasonType;
+
+typedef struct {
+    BRDisconnectReasonType type;
+    union {
+        struct {
+            int errnum;
+            char message[128 + 1];
```

Comment:
Is `message` derived from `strerror(errno)`?  If so, there could be a function `const char * BRDisconnectReasonPosixGetMessage(...) { return strerror(...); }`.  Not a big deal either way.

Reply:
Looks like it does come from `strerror()` and the existence of `message` requires some copying when filling out the `BR{Disconnect,SyncStopped)Reason` structures.